isisd/isis_pfpacket.c

```diff
@@ -121,8 +122,13 @@ static int open_packet_socket(struct isis_circuit *circuit)
 {
 	struct sockaddr_ll s_addr;
 	int fd, retval = ISIS_OK;
+	struct vrf *vrf;
+
+	vrf = vrf_lookup_by_id(circuit->interface->vrf_id);
```

Comment:
need to handle null here (log an error and return `ISIS_WARNING`). Also please rebase the PR and squash all commits into a single one.

Reply:
This is resolved. The rebase of the PR and squashing all commits into a single one is also done.
If it is correct please merge this to master. 

Thanks,